docs/appnotes/0016-authorisation-in-tams-workflows.md

```diff
@@ -0,0 +1,121 @@
+However the implementation of this can become complex and unwieldy, especially if each Source and Flow in the system has a separate set of permissions to manage and it becomes necessary to edit them all to implement a policy change.
+
+Attribute-based Access Control (ABAC) is one approach to manage this complexity, by describing permissions policies based on the attributes of resources (Sources and Flows), and if necessary, users as well.
+However full ABAC can be challenging to implement and requires a degree of organisational maturity to construct and manage stable attributes.
+This topic is well outside the scope of this document, however there exists plenty of literature and tools implementing ABAC in general terms.
+
+In practical TAMS solutions, this could look like defining a "class" attribute which can be applied to a Source or Flow, where "class" could also be thought of as "owner" or "project".
+A permissions system then defines policies that apply to those classes.
+
+For example, consider a store shared by multiple teams from the News and Sport production teams of an organisation.
+Each team have the ability to read and write their own content, and no access to the other team's content.
+However in some cases it is necessary to share a particular Source (e.g. to work on a shared story) to the other team.
+
+| Resource       | Classes | Comments |
+| -------------- | ------- | -------- |
+| Source Sport A | sport | |
+| Source Sport B | sport | |
+| Source News X  | news, sport_share | This item is shared across to Sport |
+| Source News Y  | news | |
+
+The permissions policies are then implemented as:
+
+- Users in the "sport" user group have read and write access to all content in the "sport" class
+- Users in the "news" user group have read access to all content in the "news" class
+- Users in the "sport" user group have read access to all content in the "news" class
+
+### Implementation
+
+To implement the model above, way to hold the classes in the store is needed, along with a system to store the permissions and how they map on to classes.
+
+For the latter, [Amazon Verified Permissions](https://aws.amazon.com/verified-permissions/) and [Permify](https://github.com/Permify/permify) both serve as permissions management tools.
+They allow authorisation decisions to be made by taking a set of policies defined in some domain-specific language, along with the attributes of the user (group membership) and resource (Source/Flow class), and computing whether to allow the request.
+This decision process is intended to be run inline for each request, for example at an authenticating proxy placed in front of the API server.
+
+For storing classes, an initial proof-of-concept could be built using Source and Flow tags: for example defining "special" tags such as `authz_class.news = 1` (using one tag per class to enable querying for presence of that tag, which is not possible with e.g. a comma-separated list).
```

Comment:
Can you add these tags to the known tags listing

Reply:
Good point - [bcdf5f3](https://github.com/bbc/tams/pull/115/commits/bcdf5f39afc6718618eec51d42da1ca8a67cc45c)